File: cardbattlemodule/src/main/java/com/syahputrareno975/cardbattlemodule/interfaces/RoomStreamEventController.java
package com.syahputrareno975.cardbattlemodule.interfaces;

import com.syahputrareno975.cardbattlemodule.model.card.CardModel;
import com.syahputrareno975.cardbattlemodule.model.player.PlayerModel;
import com.syahputrareno975.cardbattlemodule.model.room.RoomDataModel;

public interface RoomStreamEventController {
    void playerJoin(PlayerModel p);
    void getOneRoom(String id);
    void deployCard(PlayerModel p, CardModel c);
    void pickUpCard(PlayerModel p,CardModel c);
    void leftGame(PlayerModel p,RoomDataModel r,SimpleUnit action);
}

File: app/src/main/java/com/syahputrareno975/simplecardbattle/interfaces/RoomStreamEventController.kt
package com.syahputrareno975.simplecardbattle.interfaces

import com.syahputrareno975.simplecardbattle.model.card.CardModel
import com.syahputrareno975.simplecardbattle.model.player.PlayerModel
import com.syahputrareno975.simplecardbattle.model.room.RoomDataModel

interface RoomStreamEventController {
    fun playerJoin(p : PlayerModel)
    fun getOneRoom(id : String)
    fun deployCard(p : PlayerModel,c : CardModel)
    fun pickUpCard(p : PlayerModel,c : CardModel)
    fun leftGame(p :PlayerModel,r :RoomDataModel)
}